src/rest_api.c

```diff
@@ -89,19 +90,24 @@ static int handler(struct mg_connection * conn, void * ignored)
 
 int auth_handler(struct mg_connection * conn, void * cbdata)
 {
-    int authorized = 1;
+    int authorized = 0;
 
-    char * auth_token = mg_get_header(conn, "Authorization");
+    point_t *public_key = (point_t *)cbdata;
+
+    char * auth_token = mg_get_header(conn, "Authorization");  // TODO: Validate JWT Token
     if(auth_token != NULL){
-        ///////////////////////////////////////////////////////
-        // TODO: Validate JWT TOKEN HERE
-        ///////////////////////////////////////////////////////
-    }
+        const char *token = auth_token + 7;  // skip prefix (Bearer)
```

Comment:
Add a check that auth_token is longer than 7 bytes otherwise, the pointer will point to unallocated memory